components/applications-service/integration_test/ingestion_test.go

```diff
@@ -672,3 +673,60 @@ func TestIngestConcurrencySafe(t *testing.T) {
 	assert.Equal(t, messageNumber, suite.Ingester.EventsProcessed())
 	assert.Equal(t, messageNumber, suite.Ingester.EventsSuccessful())
 }
+
+func TestNewSgAndDeploymentUpdate(t *testing.T) {
+	defer suite.DeleteDataFromStorage()
+	suite.Ingester.ResetStats()
+
+	eventsProcessed := suite.Ingester.EventsProcessed()
+	event1 := NewHabitatEvent(
+		withSupervisorId("1"),
+		withPackageIdent("core/db/0.1.0/20200101121212"),
+		withServiceGroup("db.default"),
+	)
+	bytes1, err := proto.Marshal(event1)
+	require.NoError(t, err)
+	suite.Ingester.IngestMessage(bytes1)
+	eventsProcessed++
+
+	event2 := NewHabitatEvent(
+		withSupervisorId("2"),
+		withPackageIdent("core/db/0.1.0/20200101121212"),
+		withServiceGroup("db.default"),
+	)
+	bytes2, err := proto.Marshal(event2)
+	require.NoError(t, err)
+	suite.Ingester.IngestMessage(bytes2)
+	eventsProcessed++
+	suite.WaitForEventsToProcess(eventsProcessed)
+
+	// TODO: replace printf with test to assert we have 2 services w/ 2 supervisors in 1 service group in 1 deployment
+	svcList, err := suite.StorageClient.GetServices("name", true, 1, 5, nil)
+	require.NoError(t, err)
+	for _, svc := range svcList {
+		fmt.Printf("%+v\n", svc)
+	}
+
+	fmt.Println("-----------------------------")
+
+	// event3 changes the app_name and environment of the service from event 2
+	event3 := NewHabitatEvent(
+		withSupervisorId("2"),
+		withPackageIdent("core/db/0.1.0/20200101121212"),
+		withServiceGroup("db.default"),
+		withApplication("newApp"),
+		withEnvironment("newEnv"),
+	)
+	bytes3, err := proto.Marshal(event3)
+	require.NoError(t, err)
+	suite.Ingester.IngestMessage(bytes3)
+	eventsProcessed++
+	suite.WaitForEventsToProcess(eventsProcessed)
+
+	// TODO: replace printf with test to assert we have 2 services w/ 2 supervisors in 2 service groups in 2 deployments
+	svcList, err = suite.StorageClient.GetServices("name", true, 1, 5, nil)
+	require.NoError(t, err)
+	for _, svc := range svcList {
+		fmt.Printf("%+v\n", svc)
```

Comment:
Prior to fixing the update logic, this prints (look at app and env):

```
&{ID:48 SupMemberID:1 Origin:core Name:db Version:0.1.0 Release:20200101121212 Status: Health:OK Group:db.default Fqdn: Application:newApp Environment:newEnv Channel: Site:us PreviousHealth:NONE UpdateStrategy:NONE LastEventOccurredAt:2019-07-24 20:36:38.118062 +0000 +0000 HealthUpdatedAt:2019-07-24 20:36:38.147417 +0000 +0000}
&{ID:47 SupMemberID:2 Origin:core Name:db Version:0.1.0 Release:20200101121212 Status: Health:OK Group:db.default Fqdn: Application:newApp Environment:newEnv Channel: Site:us PreviousHealth:NONE UpdateStrategy:NONE LastEventOccurredAt:2019-07-24 20:36:38.16734 +0000 +0000 HealthUpdatedAt:2019-07-24 20:36:38.14151 +0000 +0000}
```

After fixing, it prints:

```
&{ID:51 SupMemberID:1 Origin:core Name:db Version:0.1.0 Release:20200101121212 Status: Health:OK Group:db.default Fqdn: Application:app Environment:test-env Channel: Site:us PreviousHealth:NONE UpdateStrategy:NONE LastEventOccurredAt:2019-07-24 21:31:46.952817 +0000 +0000 HealthUpdatedAt:2019-07-24 21:31:46.986629 +0000 +0000}
&{ID:50 SupMemberID:2 Origin:core Name:db Version:0.1.0 Release:20200101121212 Status: Health:OK Group:db.default Fqdn: Application:newApp Environment:newEnv Channel: Site:us PreviousHealth:NONE UpdateStrategy:NONE LastEventOccurredAt:2019-07-24 21:31:46.997118 +0000 +0000 HealthUpdatedAt:2019-07-24 21:31:46.977513 +0000 +0000}
```